# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ImageContainer.h
# Commit: 3d201ea88341
# Full Hash: 3d201ea8834161911890a06bae578b64d18eefa9
# Author: Kelsey Gilbert <jgilbert@mozilla.com>
# Date: 2021-12-15 09:11:58
# Regressor Bug: 1746111
# File Overlap Count: 1
# Description:
#   Bug 1746111 - (Re)serialize WebRTC frame SurfaceDescriptorBuffers for convert-and-upload on WebGLParent. r=gfx-reviewers,nika,lsalzman
#   
#   Also generally de-inline (WebGLParent::TexImage not DispatchCommand)
#   TexImage(SurfaceDescriptor).
#   
# ==============================================================================

diff -r a87211c8eedd -r 3d201ea88341 gfx/layers/ImageContainer.h
--- a/gfx/layers/ImageContainer.h	Wed Dec 15 01:49:11 2021 +0000
+++ b/gfx/layers/ImageContainer.h	Wed Dec 15 01:54:21 2021 +0000
@@ -659,6 +659,8 @@
   gfx::IntRect GetPictureRect() const {
     return gfx::IntRect(mPicX, mPicY, mPicSize.width, mPicSize.height);
   }
+
+  static Maybe<PlanarYCbCrData> From(const SurfaceDescriptorBuffer&);
 };
 
 // This type is currently only used for AVIF and therefore makes some